instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java

```diff
@@ -111,32 +112,43 @@ protected void onError(Span span, Throwable throwable) {
     }
   }
 
-  protected void onPeerConnection(Span span, CONNECTION connection) {
+  protected void setNetSemanticConvention(Span span, CONNECTION connection) {
     NetPeerUtils.setNetPeer(span, peerAddress(connection));
   }
 
   protected void onStatement(Span span, String statement) {
     span.setAttribute(SemanticAttributes.DB_STATEMENT, statement);
   }
 
-  // TODO: "When it's impossible to get any meaningful representation of the span name, it can be
-  // populated using the same value as db.name" (c) spec
-  protected String spanName(String query) {
-    return query == null ? DB_QUERY : query;
-  }
-
-  protected abstract String normalizeQuery(QUERY query);
+  protected abstract @NonNull String normalizeQuery(QUERY query);
 
-  protected abstract String dbSystem(CONNECTION connection);
+  protected abstract @NonNull String dbSystem(CONNECTION connection);
 
-  protected abstract String dbUser(CONNECTION connection);
+  protected String dbUser(CONNECTION connection) {
+    return null;
+  }
 
-  protected abstract String dbName(CONNECTION connection);
+  protected String dbName(CONNECTION connection) {
+    return null;
+  }
 
-  // TODO make abstract after implementing in all subclasses
   protected String dbConnectionString(CONNECTION connection) {
     return null;
   }
 
   protected abstract InetSocketAddress peerAddress(CONNECTION connection);
+
+  // TODO: "When it's impossible to get any meaningful representation of the span name, it can be
```

Comment:
I realize this is just copied, but maybe remove `TODO`, it seems to be an implementation of the spec with no todos from what I can tell.

Reply:
O yes, I have fixed this TODO in this PR. Will remove